src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/RoutePatternUsageDetector.cs

```diff
@@ -65,13 +90,26 @@ public static RoutePatternUsageContext BuildContext(SyntaxToken token, SemanticM
                 {
                     return default;
                 }
-                return new(MethodSymbol: actionMethodSymbol, MethodSyntax: methodDeclarationSyntax, IsMinimal: false, IsMvcAttribute: true);
+
+                var parameterSymbols = RoutePatternParametersDetector.GetParameterSymbols(actionMethodSymbol);
+                var resolvedParameterSymbols = RoutePatternParametersDetector.ResolvedParameters(actionMethodSymbol, wellKnownTypes);
+                return new(
+                    MethodSymbol: actionMethodSymbol,
+                    MethodSyntax: methodDeclarationSyntax,
+                    UsageType: RoutePatternUsageType.MvcAction,
+                    Parameters: parameterSymbols,
+                    ResolvedParameters: resolvedParameterSymbols);
             }
             else if (attributeParent is ClassDeclarationSyntax classDeclarationSyntax)
             {
                 var classSymbol = semanticModel.GetDeclaredSymbol(classDeclarationSyntax, cancellationToken);
-
-                return new(MethodSymbol: null, MethodSyntax: null, IsMinimal: false, IsMvcAttribute: MvcDetector.IsController(classSymbol, wellKnownTypes));
+                var usageType = MvcDetector.IsController(classSymbol, wellKnownTypes) ? RoutePatternUsageType.MvcController : RoutePatternUsageType.Other;
```

Comment:
Nit: Would it be better to just return `default` in the `Other` non-controller class case, and limit the enum to the cases we might actually act on? We don't do this for attributes on a non-action method, right?